File: 2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

void prompt_and_exit();

int main(int argc, string argv[])
{
    // Ensure that command line argument is one and digit
    if (argc != 2)
    {
        prompt_and_exit();
    }
    // Ensure that command line argument is number
    for (int i = 0, n = strlen(argv[1]); i < n; i++)
    {
        if (!isdigit(argv[1][i]))
        {
            prompt_and_exit();
        }
    }
    // Convert argv[1] to intger
    int key = atoi(argv[1]);
    // Ensure that command line argument is postive number
    if (key < 0)
    {
        prompt_and_exit();
    }
    // Ask user for the input
    string text = get_string("plaintext: ");
    // Iterate over every character in the plaintext and encrypt
    printf("ciphertext: ");
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i]))
        {
            printf("%c", 'A' + (text[i] - 'A' + key) % 26);
        }
        else if (islower(text[i]))
        {
            printf("%c", 'a' + (text[i] - 'a' + key) % 26);
        }
        else
        {
            printf("%c", text[i]);
        }
    }
    printf("\n");
    return 0;
}

void prompt_and_exit()
{
    // Prompt the user continusally for the usage
    printf("Usage: ./caesar key\n");
    exit(1);
}

File: 6/caesar/caesar.py
#!/usr/bin/env python3
# caesar

from sys import argv
from cs50 import get_string


def prompt_and_exit():
    # Prompt the user for the usage
    print('Usage: python caesar.py k')
    exit(1)


def check_key():
    # Ensure that command line argument is a digit
    if not argv[1].isdigit():
        prompt_and_exit()
    # Convert argv[1] to integer
    key = int(argv[1])
    if key < 0:  # Ensure that command line argument is positive number
        prompt_and_exit()
    return key


def main():
    # Ensure that command line argument is one
    if len(argv) != 2:
        prompt_and_exit()
    key = check_key()
    # Ask user for the input
    text = get_string("plaintext: ")
    print("ciphertext: ", end='')
    for c in text:  # Iterate over every character in the plaintext and encrypt
        if c.isupper():
            print(chr(ord('A') + (ord(c) - ord('A') + key) % 26), end='')
        elif c.islower():
            print(chr(ord('a') + (ord(c) - ord('a') + key) % 26), end='')
        else:
            print(c, end='')
    print()
    return 0


if __name__ == '__main__':
    main()
